packages/core/src/heatmap.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ISpecProp,
+  IOnboardingSpec,
+  IOnboardingMessage,
+  defaultOnboardingStages,
+  EDefaultOnboardingStages,
+  IOnboardingStage,
+  TooltipPosition,
+} from "./interfaces";
+import { getAnchor } from "./utils";
+
+export interface IOnboardingHeatmapSpec extends IOnboardingSpec {
+  chartTitle?: ISpecProp;
+  heatmapDescription?: ISpecProp;
+  legendDescription?: ISpecProp;
+  axisDescription?: ISpecProp;
+  hoverDescription?: ISpecProp;
+  missingDataDescription?: ISpecProp;
+}
+
+function generateMessages(
+  spec: IOnboardingHeatmapSpec,
+  visElement: Element
+): IOnboardingMessage[] {
+  const analyzing = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.ANALYZING
+  ) as IOnboardingStage;
+  const reading = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.READING
+  ) as IOnboardingStage;
+  const interacting = defaultOnboardingStages.get(
+    EDefaultOnboardingStages.USING
+  ) as IOnboardingStage;
+
+  const messages = [
```

Comment:
I adapted the messages and added some for anylyzing the chart:

- Reading the chart
	- 1: (title): This heatmap shows the average temperature in a week.
	- 2: (first cell): It is based on colored cells
	- 3: (legend): A deep [red] color indicates a high temperature where a deep [blue] color indicates a low temperature. Medium values (22°C - 26°C) are visualized by a neutral light gray.
	- 4: (bottom left corner (between x and y axis)): The average temperature per day time is plotted in rows and the weekday in columns.
	-
- Interacting with the chart
	- 1: (any cell): Hover over a cell to see the exact value of it.
- Analyzing the  chart
	- 1: (max value): The temperature reached it's peak on Thursday afternoon with 30°C.
	- 2: (min value): The lowest temperature was reached on Thursday evening.
	- 3: (first null value): The data for Tuesday morning is missing which is why the heatmap does not show any value for that.
	- 4: (any cell): Generally speaking, the heatmap shows that the temperature tends to be rather cool in this week (mostly blue areas), having a slight rise from Wednesday afternoon to Thursday afternoon before it drops again on Thursday in the evening.